keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -124,38 +124,71 @@ class EnvironmentPromotionChecker(
     deliveryConfig: DeliveryConfig,
     artifact: DeliveryArtifact,
     version: String,
-    targetEnvironment: String
+    targetEnvironment: Environment
```

Comment:
Every time I see us replace a string parameter with the actual object it warms my heart. ❤️ 🙂